addons/point_of_sale/static/src/js/Chrome.js

```diff
@@ -1,120 +1,141 @@
+        env.legacyActionManager = legacyActionManager;
+
+        // The proxy requires the instance of PosGlobalState to function properly.
+        env.proxy.set_pos(reactivePos);
+
+        // TODO: Should we continue on exposing posmodel as global variable?
+        // Expose only the reactive version of `pos` when in debug mode.
+        window.posmodel = pos.debug ? reactivePos : pos;
+
+        this.wowlEnv = this.env;
+        env.services.pos = this.wowlEnv.services.pos;
+        env.services.sound = this.wowlEnv.services.sound;
+        window.sound = env.services.sound;
+        this.env = env;
+        this.__owl__.childEnv = env;
+        useSubEnv({
+            get isMobile() {
+                return window.innerWidth <= 768;
+            },
+        });
+        let currentIsMobile = this.env.isMobile;
+        const updateUI = debounce(() => {
+            if (this.env.isMobile !== currentIsMobile) {
+                currentIsMobile = this.env.isMobile;
+                this.render(true);
+            }
+        }, 15); // FIXME POSREF use throttleForAnimation?
+        useExternalListener(window, "resize", updateUI);
+        onWillUnmount(updateUI.cancel);
+        // END ChromeAdapter
+
         super.setup();
         useExternalListener(window, "beforeunload", this._onBeforeUnload);
         useListener("show-main-screen", this.__showScreen);
-        useListener("toggle-debug-widget", debounce(this._toggleDebugWidget, 100));
-        useListener("toggle-mobile-searchbar", this._toggleMobileSearchBar);
         useListener("show-temp-screen", this.__showTempScreen);
         useListener("close-temp-screen", this.__closeTempScreen);
         useListener("close-pos", this._closePos);
         useListener("loading-skip-callback", () => this.env.proxy.stop_searching());
-        useListener("play-sound", this._onPlaySound);
+        const sound = useService("sound");
+        useListener("play-sound", ({ detail: name }) => sound.play(name));
         useListener("set-sync-status", this._onSetSyncStatus);
         useListener("show-notification", this._onShowNotification);
         useListener("close-notification", this._onCloseNotification);
         useListener("connect-to-proxy", this.connect_to_proxy);
         useBus(this.env.posbus, "start-cash-control", this.openCashControl);
         NumberBuffer.activate();
 
-        this.state = useState({
-            uiState: "LOADING", // 'LOADING' | 'READY' | 'CLOSING'
-            debugWidgetIsShown: true,
-            mobileSearchBarIsShown: false,
-            hasBigScrollBars: false,
-            sound: { src: null },
-            notification: {
-                isShown: false,
-                message: "",
-                duration: 2000,
-            },
-            loadingSkipButtonIsShown: false,
-        });
+        this.state = usePos();
 
-        this.mainScreen = useState({ name: null, component: null });
+        this.mainScreen = this.state.mainScreen;
```

Comment:
I wanted to avoid having to adapt overrides that use `this.mainScreen` in this PR to keep the scope in check. The plan is to move the `showScreen` method to the pos store completely in the end.